### PR TITLE
add `*ToTibble()` variants

### LINKED ISSUE #857: Revert Always return tibble for ToDataFrame functions #841
This is what I was talking about all the time - a tiblle is NOT a data frame! esqlabsR is now broken exactly because of this - some code there expected a data frame but the subsetting does not work any more because it is different between df and tibble.

This is a breaking change that is not justified.

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
- [x] `simulationResultsToTibble()`
- [x] `dataSetToTibble()`
- [x] `pkAnalysesToTibble()`
- [x] `populationToTibble()`
- [x] tests for the same

Closes #857

### CODE REVIEW

File: R/utilities-data-set.R [hunk 1]
```diff
@@ -77,27 +77,24 @@ dataSetToDataFrame <- function(dataSets) {
   dataSets <- c(dataSets)
   validateIsOfType(dataSets, "DataSet")
 
-  # styler: off
```
Comment (IndrajeetPatil) [inline R/utilities-data-set.R:80]: Like we had decided, no tags to turn off styler for any portion of the code.

File: R/utilities-data-set.R [hunk 2]
```diff
@@ -109,9 +106,19 @@ dataSetToDataFrame <- function(dataSets) {
     df[[name]] <- col
   }
 
-  return(dplyr::as_tibble(df))
+  # consistently return a (classical) data frame
+  return(as.data.frame(df))
```
Comment (PavelBal) [inline R/utilities-data-set.R:110]: Why do you use `as.data.frame` here? Isn't `df` already created as a classical df?

File: R/utilities-pk-analysis.R [hunk 3]
```diff
@@ -95,5 +95,16 @@ pkAnalysesToDataFrame <- function(pkAnalyses) {
     }
   )
 
-  return(dplyr::as_tibble(dataFrame))
+  # consistently return a (classical) data frame
+  return(as.data.frame(df))
```
Comment (PavelBal) [inline R/utilities-pk-analysis.R:99]: Same question as for  `dataSetToDataFrame`.

File: R/utilities-data-set.R (hunk 2 above)
Comment (IndrajeetPatil) [reply R/utilities-data-set.R]: Reverted.

File: R/utilities-pk-analysis.R (hunk 3 above)
Comment (IndrajeetPatil) [reply R/utilities-pk-analysis.R]: Reverted

File: R/utilities-pk-analysis.R [hunk 4]
```diff
@@ -95,5 +95,16 @@ pkAnalysesToDataFrame <- function(pkAnalyses) {
     }
   )
 
-  return(dplyr::as_tibble(dataFrame))
+  # consistently return a (classical) data frame
+  return(df)
```
Comment (PavelBal) [inline R/utilities-pk-analysis.R:99]: `df` is not present, here it is called `dataFrame`

Comment (IndrajeetPatil) [reply R/utilities-pk-analysis.R:99]: Strange that tests didn't fail.

Renamed `dataFrame` to `df` to be consistent across functions.

**Review by PavelBal — APPROVED** [review]

**Pull request merged**